pubsub.go

```diff
@@ -638,6 +660,54 @@ func (p *PubSub) handleAddSubscription(req *addSubReq) {
 	req.resp <- sub
 }
 
+// handleAddRelay adds a relay for a particular topic. If it is
+// the first relay and no subscriptions exist so far for the topic , it will
+// announce that this node relays for the topic.
+// Only called from processLoop.
+func (p *PubSub) handleAddRelay(req *addRelayReq) {
+	topic := req.topic
+
+	// announce we want this topic if neither relays nor subs exist so far
+	if p.myRelays[topic] == 0 && len(p.mySubs[topic]) == 0 {
+		p.disc.Advertise(topic)
+		p.announce(topic, true)
+		p.rt.Join(topic)
+	}
+
+	p.myRelays[topic]++
```

Comment:
We need to increment the counter _before_ trying to announce; the check should then be for the counter would be against 1.

Reply:
Note that it is not necessary a problem for the code as it is written right now, but it's subtlety we would like to avoid.